vochain/account.go

```diff
@@ -283,12 +283,19 @@ func (v *State) ConsumeFaucetPayload(from common.Address, faucetPayload *models.
 	}
 
 	// transfer tx cost to burn address
-	collectFaucetCost, err := v.TxCost(models.TxType_COLLECT_FAUCET, false)
-	if err != nil {
-		return err
-	}
-	if err := accFrom.Transfer(burnAcc, collectFaucetCost); err != nil {
-		return fmt.Errorf("cannot transfer balance: %w", err)
+	// if tx is set account info (create fresh account) burn tokens here
+	// other case is collect faucet Tx where tokens are burned in a different function
+	// if tx is not set account (no new account) increment nonce of the tx sender
+	if txType == models.TxType_SET_ACCOUNT_INFO {
+		collectFaucetCost, err := v.TxCost(models.TxType_COLLECT_FAUCET, false)
+		if err != nil {
+			return err
+		}
+		if err := accFrom.Transfer(burnAcc, collectFaucetCost); err != nil {
+			return fmt.Errorf("cannot transfer balance: %w", err)
```

Comment:
wouldn't this be "cannot burn fee"? transfer balance sounds confusing to me, but maybe i'm just lacking context.

Reply:
mm not sure, because you are transferring to the burn address but .. thinking a bit more i think is more useful to talk about burn fee in order to have more self explanatory error.

Reply:
yup, added a better error :) thank you